### DISCUSSION TITLE
Filtering out IfcOpeningElement's in applicability statements.

### DISCUSSION THREAD

**Discussion by Xema-Pastor** [buildingSMART/IDS#376] [Q&A] [answered]
Hello.

We have a problem with voids in walls, when we try to check a specification like "All physical elements must have certain property".

We are using REVIT in our native models. We build the wall, insert a door or a window, and assign them the desired property in a PSET. When we export the model to ifc, an IfcOpeningElement is automatically created within the wall, but this IfcOpeningElement does not inherit the Property assigned to the wall. It is "empty", clear of any bound to the PSET

We consider "physical elements" those having the ObjectType attribute:

```
<ids:applicability minOccurs="1" maxOccurs="unbounded">
    <ids:attribute>
        <ids:name>
            <ids:simpleValue>
                ObjectType
            </ids:simpleValue>
        </ids:name>
    </ids:attribute>
</ids:applicability>
```
IfcOpeningElement's match this filter and are checked. As a result, we get a report with "a bunch of element does not fulfill the requirement", and it is not true, the model is properly built.

So, my question is: Is there a way to filter out the IfcOpeningElement's in the Applicability filter? Something like "get all this elements, but no this sub-set of them".
We could simply re-make the Applicability filter with an explicit ennumeration, something like "search IfcWall, IfcWindow, IfcColumn,...", but we think the "logical way" to collect and reject elements is preferred. Or, perhaps, the way to collect the physical elements with the ObjectType attibute is not a good idea.

If the "logical way" of filter out subsets is not supported by the IDS schema right now, please, consider to include it in future versions of the standar.

Thank you in advance.

**Comment by atomczak** [comment, marked as answer]
Hello! This is often a repeated request, and we did consider it. I don't recommend the approach with the attribute ObjectType, as you get all objects, including non-physical ones like IfcBuildingStorey. Instead, I would indeed be specific and list all physical entities that you are interested in. This way, the person interpreting your specification doesn't need to know exactly what elements you have in mind. We might need to think of a better solution in the future.

**Reply by Xema-Pastor** [reply]
Understood, Artur, thank you so much
